tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/mqtt5/MQTT5Test.java

```diff
@@ -871,4 +875,53 @@ public void testFlowControlAfterDisconnect() throws Exception {
       Wait.assertEquals(1L, () -> subscriptionQueue.getMessagesAcknowledged(), 2000, 50);
       Wait.assertEquals(0L, () -> subscriptionQueue.getMessageCount(), 2000, 50);
    }
+
+   /**
+    * This test helps validate that there are no longer timeouts related to persisting subscriptions under load.
+    * However, it is disabled because it takes a relatively long time to run. I'm leaving it here for ad hoc runs
+    * as necessary.
+    */
+   @Disabled
+   @Test
+   @Timeout(DEFAULT_TIMEOUT_SEC)
+   public void testLoadOnSubscriptionPersistence() throws Exception {
+      String topic = RandomUtil.randomUUIDString();
+      int numberOfThreads = 250;
+      int numberOfSubscriptions = 1000;
+      AtomicBoolean failed = new AtomicBoolean(false);
+      ExecutorService executorService = Executors.newFixedThreadPool(numberOfThreads);
+      runAfter(executorService::shutdownNow);
```

Comment:
@jbertram beware that I pushed -f into your fork with a few changes here...

runAfter the executorService
and countdownLatch for the completion task...


Can you rebase please?

Reply:
pushed another change.. just one subscription